src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidationContext.cs

```diff
@@ -69,6 +68,6 @@ public class OpenIdConnectProtocolValidationContext
         /// <summary>
         /// This id_token is assumed to have audience, issuer, lifetime and signature validated.
         /// </summary>
-        public JwtSecurityToken ValidatedIdToken { get; set; }
+        public SecurityToken ValidatedIdToken { get; set; }
```

Comment:
I'm wondering if ValidateIdToken should be JsonWebToken.
We are treating ValidateIdToken as JsonWebToken:

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/607218ba549e4a00b8f8a078c25a1e76e7ce019d/src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs#L296

Same comment applies to IdTokenValidator.

Reply:
That's a good point. Since it's always guaranteed to be a JsonWebToken that might make sense.

I was thinking that SecurityToken might be more future-proof as we wouldn't need to update the property type if we decided to use a different token type in the future. However, I'm not sure how likely this scenario is.

We actually do something similar in TokenValidationResult:

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/7c7c7e13968e6656e67eb43938d4da136336049c/src/Microsoft.IdentityModel.Tokens/TokenValidationResult.cs#L58-L61

We always cast SecurityToken to a JsonWebToken since it's only ever used by the JsonWebTokenHandler.